src/Components/Topics.js

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect } from "react";
+import { fetchTopics } from "../api";
+import { Link } from "react-router-dom";
+
+export default function Topics() {
+  const [topics, setTopics] = useState([]);
+
+  useEffect(() => {
+    fetchTopics().then(({ data }) => {
+      setTopics(data.data);
+    });
+  }, []);
+
+  const handleTopic = (event) => {
+    setTopics(event.target.value);
```

Comment:
I think you need to have another think about what's happening here. 
You've already fetched and set all the topics in tour useEffect. Now you're re-setting the topics to be whatever value the current button being clicked is - which will only be ONE topic - yet seems to be being used to set ALL topics, which is confusing and seems to be mixing up your functionality a bit.

If I run your repo and click an individual topic, it does something, and then quickly re-thinks it to change the info the user is seeing (you have to be eagle-eyed, but it's def happening) - I'm not sure if that's down to what I've described above, but it's probably what I'd put my money on.